guides/specification.md

```diff
@@ -0,0 +1,555 @@
+  `values.some_value` is `{{ values.some_value }}`
+  `env.some_env` is `{{ env.some_env }}`
+
+description: |
+  Corosync `token` timeout is set to `{{ values.expected_token_timeout }}`
+```
+
+#### remediation
+
+A text providing an comprehensive description about the remediation to apply for the Check being declared.
+
+It has the same properties of the `description`
+- can be a one-liner (it usually is not)
+- can be a multiline (it usually is)
+- format is **markdown**
+- has access to [`env`](#env) and [`values`](#values-1) evaluation scopes (not yet true, and not yet defined if needed)
+
+Example:
+```yaml
+remediation: |
+  ## Abstract
+  The value of the Corosync `token` timeout is not set as recommended.
+  ## Remediation
+  Adjust the corosync `token` timeout as recommended on the best 
+  ...
+  2. Reload the corosync configuration:
+  ...
+
+```
+
+#### Severity
+
+A string determining the severity of the Check being declared, in case the check is not passing, so that the appropriate result is reported.
+
+Allowed values: `warning`, `critical`
+
+**Default:** if no severity is provided, the system would default to `critical`
+
+Example:
+
+Reports a `warning` When the Check expectations do not pass
+```yaml
+severity: warning
+```
+
+Reports a `critical` When the Check expectations do not pass
+```yaml
+severity: critical
+```
+
+#### Facts, Values, Expectations
+
+See main sections [Facts](#facts), [Values](#values), [Expectations](#expectations)
+
+## Facts
+
+Facts are various types of information that the engine can discover on the target infrastructure.
+Examples include (but are not limited to) installed packages, open ports, and configuration files' content.
+
+The process of determining the value of a declared fact during Check execution is refered to as _Facts Gathering_ and it is the job of the [_Gatherers_](./gatherers.md).
+Think of gatherers as being functions that have a name and accept argument(s).
+
+That said, a fact declaration contains:
+- the fact name
+- the gatherer used to retrieve the fact
+- the argument(s) to be provided to the gatherer
+
+**Note:** 
+- many facts can be declared
+- all the declared facts would be registered in the [`facts`](#facts-1) namespaced evaluation scope.
+
+```yaml
+facts:
+  - name: <fact_name>
+    gatherer: <gatherer_name>
+    argument: <gatherer_argument>
+    
+  - name: <another_fact_name>
+    gatherer: <another_gatherer_name>
+    argument: <another_gatherer_argument>
+```
+
+The following example declares a **fact** named `corosync_token_timeout`, retrievable via the built-in `corosync.conf` **gatherer** to which will be provided the **argument** `totem.token` 
+```yaml
+facts:
+  - name: corosync_token_timeout
+    gatherer: corosync.conf
+    argument: totem.token
+
+  # other facts maybe
+```
+
+Finally, gathered facts, will be used in Check's [Expectations](#expectations) to determine whether expected conditions are met for the best practice to be adhered.
```

Comment:
Let's try to not use future term. Something like `Finally, gathered facts, are used in...`